src/runtime/runtime.js

```diff
@@ -816,6 +824,31 @@ export class ConfiguredRuntime {
     this.buttonApi_.attach(button, optionsOrCallback, opt_callback);
   }
 
+  /** @override */
+  createSubscriptionButton(container, options) {
+    const opts = /** @type {!ButtonOptions} */
+        (Object.assign(options || {}, {theme: Theme.LIGHT}));
+    if (!isExperimentOn(this.win_, ExperimentFlags.SMARTBOX)) {
+      throw new Error('Not yet launched!');
+    }
+    if (!container) {
+      throw new Error('No element found to render Smart button!');
+    }
+    return this.documentParsed_.then(() => {
+      return this.getEntitlements().then(entitlements => {
```

Comment:
`getEntitlements` API call is required to find if there are `subscriptions` and if any, do not display `smartbutton.
Also, `isReadyToPay` flag is passed to the `SmartSubscriptionButtonFlow` to decide what `message` is rendered underneath the button.